stdlib/public/core/Substring.swift.gyb

```diff
@@ -612,3 +612,42 @@ extension Substring {
   }
 }
 //===----------------------------------------------------------------------===//
+
+// popFirst() is only present when a collection is its own subsequence. This was
+// dropped in Swift 4.
+extension String {
+  @available(swift, deprecated: 3.2)
```

Comment:
The problem is that this useful message will not be displayed as part of deprecation warning.
It is possible to combine `deprecated` and `obsoleted` in a single attribute though, this way the message will be shared.